src/Jobs/Recruit.Vacancies.Jobs/ExternalSystemEventHandlers/UpdatedPermissionsExternalSystemEventsHandler.cs

```diff
@@ -0,0 +1,94 @@
+        private readonly IEmployerAccountProvider _employerAccountProvider;
+        private readonly IEncodingService _encoder;
+        private readonly RecruitWebJobsSystemConfiguration _jobsConfig;
+        private readonly IVacancyQuery _vacancyQuery;
+        private readonly IMessaging _messaging;
+        private string ExternalSystemEventHandlerName => GetType().Name;
+
+        public UpdatedPermissionsExternalSystemEventsHandler(ILogger<UpdatedPermissionsExternalSystemEventsHandler> logger, RecruitWebJobsSystemConfiguration jobsConfig,
+                                                IRecruitQueueService recruitQueueService,
+                                                IEmployerAccountProvider employerAccountProvider, IEncodingService encoder, IVacancyQuery vacancyQuery,
+                                                IMessaging messaging)
+        {
+            _logger = logger;
+            _jobsConfig = jobsConfig;
+            _recruitQueueService = recruitQueueService;
+            _employerAccountProvider = employerAccountProvider;
+            _encoder = encoder;
+            _vacancyQuery = vacancyQuery;
+            _messaging = messaging;
+        }
+
+        public async Task Handle(UpdatedPermissionsEvent message, IMessageHandlerContext context)
+        {
+            if (_jobsConfig.DisabledJobs.Contains(ExternalSystemEventHandlerName))
+            {
+                _logger.LogDebug($"{ExternalSystemEventHandlerName} is disabled, skipping ...");
+                return;
+            }
+
+            if (message.UserRef == Guid.Empty)
+            {
+                _logger.LogInformation($"Not handling Provider {nameof(Operation.Recruitment)} Permission being revoked as it is a consequence of Provider being blocked by QA on Recruit.");
+                return;
+            }
+
+            if (message.GrantedOperations.Contains(Operation.Recruitment) == false)
+            {
+                var legalEntity = await GetAssociatedLegalEntity(message);
+
+                if (legalEntity == null)
+                {
+                    throw new Exception($"Could not find matching Account Legal Entity Id {message.AccountLegalEntityId} for Employer Account {message.AccountId}");
+                }
+
+                var noOfAssociatedVacancies = (await _vacancyQuery.GetNoOfProviderOwnedVacanciesForLegalEntityAsync(message.Ukprn, legalEntity.LegalEntityId));
+
+                if (noOfAssociatedVacancies > 0)
+                {
+                    await _recruitQueueService.AddMessageAsync(new TransferVacanciesFromProviderQueueMessage
+                    {
+                        Ukprn = message.Ukprn,
+                        LegalEntityId = legalEntity.LegalEntityId,
+                        UserRef = message.UserRef,
+                        UserEmailAddress = "",
+                        UserName = "" // concat (awaiting events update in provider relationships)
```

Comment:
Can we set the transfer reason at this point on the message? Otherwise we have to infer it later on.  This would be more explicit.
`TransferReason = TransferReason.EmployerRevokedProviderPermission`

Reply:
See above.

Reply:
👍